src/Pages/Checkout/index.js

```diff
@@ -16,20 +28,76 @@ const CartDetail = ({ className, title, price, qty }) => {
 };
 
 const Checkout = () => {
+  const { productState, productDispatch } = useProduct();
+
+  useEffect(() => {
+    wrapAsync(async () => {
+      productDispatch({
+        type: 'ADD_TO_CART',
+        payload: await fetchCartProducts(),
+      });
+    })();
+  }, [productState.cartProducts]);
+
+  const handleIncrement = wrapAsync(async (product) => {
+    await updateProducts(product, 'increment');
+  });
+
+  const handleDecrement = wrapAsync(async (product) => {
+    if (product.qty === 1) {
+      const cartProducts = await removeProductsFromCart(product, 'decrement');
+      productDispatch({
+        type: 'ADD_TO_CART',
```

Comment:
Here your productDispatch type is ADD_TO_CART, while the function is to to decrement product quantity. It causing confusion what the function does.
You can use better naming convention here.
You can use something like SET_CART_LIST.